tests/test_tools.sh

```diff
@@ -327,38 +368,82 @@ run_command "gt_mpi_gather -A "non-existent-array" $WORKSPACE/loader.json -j $QU
 run_command "gt_mpi_gather -l $WORKSPACE/loader.json -j $QUERY_JSON --print-AC"
 run_command "gt_mpi_gather -l $WORKSPACE/loader.json -j $QUERY_JSON --print-calls"
 
-# Fail if there is no reference genome file specified
-run_command "gt_mpi_gather -l $WORKSPACE/loader.json -j $QUERY_JSON --produce-Broad-GVCF" ERR
+# Diff results from two gt_mpi_gather invocations
+# $1 Results from 1st invocation
+# $2 Results from 2nd invocation
+# $3 Expected Result(OK or ERR)
+# $4 Error Message
+diff_gt_mpi_gather_output() {
+  sed "1 d" $1 > $1.cut
+  sed -i -e '/MD5/d' $1.cut
+  sed "1 d" $2 > $2.cut
+  sed -i -e '/MD5/d' $2.cut
+  DIFF_RESULT=$(diff $1.cut $2.cut)
+  if [[ ! -z  $DIFF_RESULT && $3 == OK ]]; then
+    echo "gt_mpi_gather results differ : $4"
+    echo "First output:"
+    cat $1.cut
+    echo "Second output:"
+    cat $2.cut
+    STATUS=1
+  elif [[ -z $DIFF_RESULT && $3 == ERR ]]; then
+    echo "gt_mpi_gather results should differ but were identical : $4"
+    STATUS=1
+  fi
+}
+
+# $1 : intervals
+test_bypass_intersecting_intervals_phase() {
+  create_query_json_with_intervals "$1"
+  time_command "gt_mpi_gather -l $WORKSPACE/loader.json -j $QUERY_JSON --print-AC"
+  DIFF=$TIME
+  mv -f $TEMP_DIR/output $TEMP_DIR/output.without_bypass
+  create_query_json_with_intervals "$1" "\"bypass_intersecting_intervals_phase\": true"
+  time_command "gt_mpi_gather -l $WORKSPACE/loader.json -j $QUERY_JSON --print-AC"
+  if [ $DIFF -lt $TIME ]; then
+    echo "Bypass intersecting intervals phase took longer than the normal two pass iterator"
+    STATUS=1
+  fi
+}
+
+# Test bypass of intersecting intervals phase in the genomicsdb iterators
+INTERVALS="\"query_contig_intervals\": [{\"contig\": \"1\", \"begin\": 1, \"end\":100 }]"
+test_bypass_intersecting_intervals_phase "$INTERVALS"
+diff_gt_mpi_gather_output $TEMP_DIR/output.without_bypass $TEMP_DIR/output OK "Setting bypass intersecting intervals should not affect gt_mpi_gather for $INTERVALS for t0.vcf.gz"
+
+# Test bypass of intersecting intervals phase in the genomicsds iterators with two intervals
+INTERVALS="\"query_contig_intervals\": [{\"contig\": \"1\", \"begin\": 1, \"end\":100 }, {\"contig\": \"1\", \"begin\": 17385 }]"
```

Comment:
Maybe change the first interval to end at `12142` so that the first interval isn't empty.